crates/pep508-rs/src/lib.rs

```diff
@@ -670,18 +675,47 @@ fn looks_like_unnamed_requirement(cursor: &mut Cursor) -> bool {
     }
 
     // Ex) `https://` or `C:`
-    if split_scheme(&expanded).is_some() {
+    if split_scheme(url).is_some() {
         return true;
     }
 
     // Ex) `foo/bar`
-    if expanded.contains('/') || expanded.contains('\\') {
+    if url.contains('/') || url.contains('\\') {
+        return true;
+    }
+
+    // Ex) `foo.tar.gz`
+    if looks_like_archive(url) {
         return true;
     }
 
     false
 }
 
+/// Returns `true` if a file looks like an archive.
+///
+/// See <https://github.com/pypa/pip/blob/111eed14b6e9fba7c78a5ec2b7594812d17b5d2b/src/pip/_internal/utils/filetypes.py#L8>
+/// for the list of supported archive extensions.
+fn looks_like_archive(file: impl AsRef<Path>) -> bool {
```

Comment:
Should we use the `DistFilename` parsing here or are you intentionally using a wider filter?

Reply:
Intentionally a bit wider to match pip.